src/firebolt/async_db/connection.py

```diff
@@ -64,11 +64,43 @@ async def _resolve_engine_url(
             raise InterfaceError(f"Unable to retrieve engine endpoint: {e}.")
 
 
+def _validate_engine_name_and_url(
+    engine_name: Optional[str], engine_url: Optional[str]
+) -> None:
+    if engine_name and engine_url:
+        raise InterfaceError(
+            "Both engine_name and engine_url are provided. Provide only one to connect."
+        )
+    if not engine_name and not engine_url:
+        raise InterfaceError(
+            "Neither engine_name nor engine_url is provided. Provide one to connect."
+        )
+
+
+def _get_auth(
+    username: Optional[str], password: Optional[str], access_token: Optional[str]
+) -> AuthTypes:
+    if not access_token:
+        if not username or not password:
+            raise InterfaceError(
+                "Neither username/password nor access_token are provided. Provide one"
+                " to authenticate"
+            )
+        return (username, password)
```

Comment:
Should we create an Auth instance here with user/password instead of passing a tuple?

Reply:
To create an Auth instance here we would need to also know `api_endpoint` here. In case of tuple, it's inherited from Client

Reply:
I don't have a strong opinion in here, just thought it would look neater.